src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/OpenSslHelpers.cs

```diff
@@ -86,32 +86,38 @@ public static AlgorithmIdentifier ReadAlgoIdFromEncryptedContentInfo(this DerSeq
                 "Expected to skip an OID while reading EncryptedContentInfo");
             encryptedContentInfo.SkipValue();
 
-            return encryptedContentInfo.ReadAlgoIdentifier();
-        }
+            byte[] contentEncryptionAlgorithmIdentifier = encryptedContentInfo.ReadNextEncodedValue();
 
-        public static AlgorithmIdentifier ReadAlgoIdentifier(this DerSequenceReader encryptedContentInfo)
-        {
-            // The encoding for a ContentEncryptionAlgorithmIdentifier is just a sequence of the OID and the parameters, 
-            // but we just need the OID 
-            DerSequenceReader contentEncryptionAlgorithmIdentifier = encryptedContentInfo.ReadSequence();
-            string algoOid = contentEncryptionAlgorithmIdentifier.ReadOidAsString();
-            int keyLength = 0;
+            DerSequenceReader encryptionAlgorithmReader = new DerSequenceReader(contentEncryptionAlgorithmIdentifier);
```

Comment:
Why are you calling ReadNextEncodedValue then making a DerSequenceReader over it? That should be the same as ReadSequence()... I think?